### DISCUSSION TITLE
Package with MRTK as a dependency

### DISCUSSION THREAD

**Discussion by ahmed-shariff** [microsoft/MixedRealityToolkit-Unity#11110] [Q&A] [answered]
I am hoping this would be the right channel to ask this. I am trying to write a unity package with MRTK as a dependency. Is there a standard/recommended way of doing this? 

From my understanding of how packages and MRTK works, I would have to have the MRTK tgz files that the installation tools drops into the packages folder be packaged into it? Is there a better/more-correct way of doing that?

**Comment by Zee2** [comment, marked as answer]
Yep. For now, we recommend using the MR Feature Tool to pull down the tarballs. You can either check in these tarballs to your repository, or you can run MR Feature Tool's Restore functionality on new checkouts, which reads your package manifest and pulls down the relevant MR tarballs to your package folder.

We're currently working on improving the MR Feature Tool to offer command-line `mrft restore` functionality. In addition, we're talking with Unity to see if we are allowed to use their Scoped Registry feature for public distribution (currently, their terms of service implies that this is against the rules!)

See **Path 2** here in the setup docs https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/setup.
